presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/UnaliasSymbolReferences.java

```diff
@@ -518,7 +518,7 @@ public PlanNode visitJoin(JoinNode node, RewriteContext<Void> context)
                         .forEach(clause -> map(clause.getRight(), clause.getLeft()));
             }
 
-            return new JoinNode(node.getId(), node.getType(), left, right, canonicalCriteria, canonicalFilter, canonicalLeftHashSymbol, canonicalRightHashSymbol);
+            return new JoinNode(node.getId(), node.getType(), left, right, canonicalCriteria, canonicalizeAndDistinct(node.getOutputSymbols()), canonicalFilter, canonicalLeftHashSymbol, canonicalRightHashSymbol);
```

Comment:
If the join is cross join type, then you cannot use `canonicalizeAndDistinct` on output symbols, but you should use left and right input symbols instead.

Extract join node output enforcing into separate method.